components/list/src/provider/mod.rs

```diff
@@ -69,6 +69,11 @@ data_marker!(
     ListFormatterPatterns<'static>,
 );
 
+icu_provider::data_struct_new!(
+    ListFormatterPatterns<'_>,
+    #[cfg(feature = "datagen")]
```

Comment:
why is this setting a feature, but the other invocations are not?

Reply:
It can be applied anywhere such a feature exists. Fixed.